### ISSUE TITLE
Improve SMTP delivery reliability

### ISSUE THREAD

**Issue by louischan-oursky** [authgear/authgear-server#800]
We are now using https://github.com/go-gomail/gomail to communicate with a remote SMTP server directly.
Ben suggested we can have a local SMTP server to handle logs, retry, etc.

**Comment by chpapa** [comment]
Main thing is for logging and retry which is common to debug...

**Comment by kiootic** [comment]
I think the suggest 'SMTP server' is just to have a message queue to handle resiliency. In general, every message sending/batch operations should have this queue (e.g. web-hook events), which is NYI.

**Comment by chpapa** [comment]
yes it was just my exp the local smtp server handle it too well and easy for debug on production so my rule of thumbs was always have a local smtp server to handle all these.

Technically you can also have a common libraries that do the logging and retry really well and handle every kind of external system calls.
